app/controllers/concerns/effective_user.rb

```diff
@@ -0,0 +1,9 @@
+module EffectiveUser
+  def effective_user_id
+    [
+      session[:ial2_recovery_user_id],
+      session[:doc_capture_user_id],
+      current_user&.id,
+    ].find(&:present?)
```

Comment:
This might be a bit too "clever" in the negative sense, where something more like `SessionErrorsController#user_id` could work just as well:

```suggestion
    ial2_recovery_user_id = session[:ial2_recovery_user_id]
    return ial2_recovery_user_id if ial2_recovery_user_id.present?

    doc_capture_user_id = session[:doc_capture_user_id]
    return doc_capture_user_id if doc_capture_user_id.present?

    return current_user.id if current_user
```

Reply:
I think the way as it exists is pretty clear, I wouldn't both expanding it